### PR TITLE
[Janitorial] Use small variant of Badge component in <PlanFeaturesHeader />

### PR DESCRIPTION
**Pull request by adamziel**
This is a follow up on the following discussion: https://github.com/Automattic/wp-calypso/pull/26978#issuecomment-419400483

This PR adds a small version of the <Badge /> component and uses it in <PlanFeaturesHeader />.

Test plan:
1. As a personal plan user, go to /plans and confirms that "Credit applied" badges look pretty and scale well:

zrzut ekranu 2018-09-19 o 14 19 38

2. Go to `/devdocs/design` and confirm the preview looks like this:

zrzut ekranu 2018-09-19 o 14 20 16

### CODE REVIEW

**Comment by adamziel** [comment]
@jancavan I noticed some tests failed so I just updated this PR

**Comment by jancavan** [comment]
I think visually this looks good, nice work 👍 

Re naming convention: Do you think we need to name the bigger badge `regular`? And what do you think about using `compact` instead of `small`?

I think for cards and buttons, the bigger ones aren't named `regular` and the smaller ones are named `compact`.

File: client/components/badge/docs/example.jsx [hunk 1]
```diff
@@ -21,6 +21,12 @@ BadgeExample.defaultProps = {
 		<div>
 			<Badge type="success">Success Badge</Badge>
 			<Badge type="warning">Warning Badge</Badge>
+			<Badge type="success" size="small">
```
Comment (bluefuton) [inline client/components/badge/docs/example.jsx]: Can we use `compact` here like the Button component? For example:

`<Badge type="success" compact>`

**Comment by bluefuton** [comment]
Great idea. Happy to provide reviews to get this to the finish line 🏁

**Pull request closed without merging**